sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -582,9 +583,29 @@ class SparkSqlAstBuilder extends AstBuilder {
   }
 
   override def visitSetVariable(ctx: SetVariableContext): LogicalPlan = withOrigin(ctx) {
-    val variableName = visitMultipartIdentifier(ctx.multipartIdentifier)
-    val expression = ctx.expression().toString
-    SetVariableCommand(VariableIdentifier(variableName.last), expression)
+    if (ctx.assignmentList().isEmpty) {
+      SetVariableCommand(Seq(VariableIdentifier("var")), "7")
```

Comment:
7?! What does this "magic number" represent?

Reply:
This was temporary code